capstone/render.md

```diff
@@ -0,0 +1,244 @@
+
+
+## Deploy on Render
+
+First, we will go to [render.com](https://render.com/). Using our team's email and password, we will log into the team's accout. 
+
+### Postgres
+
+Once we are logged in, we will create a new PostgreSQL Database
+
+![ new_postgres ](./assets/new_postgres.png)
+
+- Let's make sure that we have selected the Free Plan
+- Once the database is created we will need to copy the Database URL to be used later 
+
+
+### Web Service
+
+Our next step is to create a new Web Service, pointing it to your application repository (make sure Render has access to it). 
+
+![ connect_repo ](./assets/connect_repo.png)
+
+The language we will select for our environment will be `Ruby` and then we will set the following propertires: 
+
+- Build command: `./bin/render-build.sh`
+- Start command: `bundle exec puma -C config/puma.rb`
+
+![ web_service ](./assets/web_service.png)
+
+Now, we need to add the following environment variables under the **Advanced** section: 
+
+![ web_service ](./assets/environment_variables.png)
+
+That's it! We can now finalize the backend service deployment. It will be live on our `.onrender.com` URL as soon as the build finishes. Copy and save this URL. We will need it for the front end. 
+
+### Frontend
+
+- Now that we have our URL from the backend service we just deployed, we need to change the URL in our fetch requests 
+
+- From the Render Dashboard, we will click the 'New' button and select 'Static Site'
+
+- Using our existing frontend repo, we will give Render permission to access it 
+
+- Use the following values in the creation process: 
+
+![ web_service ](./assets/creation_commands.png)
+
+- Click on 'Create Static Site'
+
+- Once our static site is deployed, we need to copy the site URL and save it for a later step. 
+
+- Because we are using React Router, we will need to direct all routing request to `index.html` so they can be handled by your routing library. 
+
+We can do this by defining a **Rewrite Rule** for our site. Go to the **Redirects/Rewrites** tab for your service and add a rule with the following values and then Save: 
+
+![ web_service ](./assets/redirects_rewrites.png)
+
+Just like that, the frontend is complete!
+
+### Final Step - allowing backend to handle frontend requests:
+
+In the `cors.rb` file in our backend, we will need to change the origins to be the url to your frontend app **MINUS** the `https://` 
+
+```ruby
+Rails.application.config.middleware.insert_before 0, Rack::Cors do
+  allow do
+    origins 'your-app-frontend.onrender.com' <--- change this line!
+    resource '*',
+    headers: ["Authorization"],
+    expose: ["Authorization"],
+    methods: [:get, :post, :put, :patch, :delete, :options, :head],
+    max_age: 600
+  end
+end
+```
+
+git add, commit, and push. Redeploy!
+
+### Troubleshooting
+During deployment, our build might encounter some errors. 
```

Comment:
It is very common during deployment for the build to encounter errors. Small syntax errors that work in development can cause a build to fail. Listed below are a few common errors we see along with the next steps for troubleshooting.